defopt.py

```diff
@@ -118,8 +118,23 @@ def _create_parser(funcs, *args, **kwargs):
     return parser
 
 
-class _Formatter(RawTextHelpFormatter, ArgumentDefaultsHelpFormatter):
-    pass
+class _Formatter(RawTextHelpFormatter):
+
+    # modified from ArgumentDefaultsHelpFormatter to add type information, and
+    # remove defaults for varargs (`ZERO_OR_MORE`).
+    def _get_help_string(self, action):
+        info = []
+        if action.type is not None and '%(type)' not in action.help:
+            info.append('%(type)s')
+        if (action.default != argparse.SUPPRESS
```

Comment:
re: "is not": these objects are strings, and it is well known that strings should be compared with =, not with "is".  I would guess that the current implementation only works because short strings are interned.
Added a note for OPTIONAL.